tethys_layouts/static/tethys_layouts/map_layout/map_layout_cesium.js

```diff
@@ -0,0 +1,2062 @@
+/*****************************************************************************
```

Comment:
What's the relationship between this and the `cesium_map_view.js` file? Do you think it would be possible/a good idea to ever merge the two files?

Reply:
Same question applies to the map_layout.js

Reply:
The `*_layout.js` files handle the Map Layout specific logic like creating and managing the layer picker/tree, legends, plot slide sheet, etc. They interact with the maps created by the `*_map_view.js` scripts, but they have distinct differences in purpose. I don't think it makes sense to merge them. If anything, we may want to split parts of the `*_layout.js` scripts out into separate Gizmos (like a Layer picker/tree Gizmos).

Reply:
Anticipating a follow-up question: work that needs to be done in the future is to find a more elegant solution for the `*_layout.js` scripts. `map_layout_cesium.js` started as a copy of the `map_layout.js` with the internals of some of the methods replaced to interact with a Cesium Map instead of an Open Layers map. For example the layer picker uses a method to toggle the visibility of layers. Using inheritance to handle this would be more elegant, but isn't an option with our enclosure module style libraries at the moment. Something to tackle in the future. The other thing is that the Cesium layouts functionality is not fully implemented and the OpenLayers logic or TODOs are found throughout. I ought to add a note to the docs that the Cesium functionality is experimental if I haven't already.